@using Serilog.Context
@inherits ErrorBoundary
@inject ILogger<LoggedErrorBoundary> _logger
@inject NavigationManager _nav

@if (CurrentException is null)
{
	@ChildContent
}
else if (ErrorContent is not null)
{
	@ErrorContent(CurrentException)
}

@code {
	protected override Task OnErrorAsync(Exception ex)
	{
		using (LogContext.PushProperty("Url", _nav.Uri))
		{
			_logger.LogError(ex, "Caught an unhandled exception!");
		}
		return Task.CompletedTask;
	}
}